### PR TITLE
Sample for assistants function calling with Bing Search

### PR DESCRIPTION
**Pull request by hkesavam**
# Description

Sample for assistants function calling with Bing Search

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure-Samples/azureai-samples/blob/main/CONTRIBUTING.md)
- [ ] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [ ] This notebook or file is added to the [CODEOWNERS](https://github.com/Azure-Samples/azureai-samples/blob/main/.github/CODEOWNERS) file, pointing to the author or the author's team.

### CODE REVIEW

**Review by diondrapeck — APPROVED** [review]

**Comment by hkesavam** [comment]
> @hkesavam please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

**Comment by kdestin** [comment]
To address the CI failure, please consult the [Contributing Guidelines](https://github.com/Azure-Samples/azureai-samples/blob/main/CONTRIBUTING.md#2-install-dev-dependencies) to install the dev requirements and `pre-commit`. Then run `pre-commit run --all-files` from the root of the repository. 

This'll:

* Format the notebooks with black
* Remove unneeded metadata
* Surface some linting issues that need to be addressed.

File: notebooks/assistants/function_calling/assistants_function_calling_with_bing_search.ipynb [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+    "### Define a function to call the Bing Search APIs\n",
+    "To learn more about using the Bing Search APIs with Azure OpenAI, see [Bing Search APIs, with your LLM](https://learn.microsoft.com/bing/search-apis/bing-web-search/use-display-requirements-llm)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 43,
+   "metadata": {},
+   "outputs": [],
+   "source": [
```
Comment (kdestin) [inline notebooks/assistants/function_calling/assistants_function_calling_with_bing_search.ipynb]: Moves user specified values into a "[parameters block](https://papermill.readthedocs.io/en/latest/usage-parameterize.html)" per the template.

```suggestion
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Parameters\n",
    "Update the following config to include details of your Azure OpenAI and Bing Search resources"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "deployment_name = \"<DEPLOYMENT_NAME>\"\n",
    "aoai_api_base = \"https://<YOUR_RESOURCE_NAME>.openai.azure.com\"\n",
    "aoai_api_version = \"2024-02-15-preview\"\n",
    "aoai_api_key = \"<AOAI_RESOURCE_API_KEY>\"\n",
    "bing_search_subscription_key = \"<BING_SEARCH_SUBSCRIPTION_KEY>\"\n",
    "bing_search_url = \"https://api.bing.microsoft.com/v7.0/search\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define a function to call the Bing Search APIs\n",
    "To learn more about using the Bing Search APIs with Azure OpenAI, see [Bing Search APIs, with your LLM](https://learn.microsoft.com/bing/search-apis/bing-web-search/use-display-requirements-llm)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import json\n",
    "import time\n",
    "\n",
    "from openai import AzureOpenAI\n",
    "\n",
    "\n",

```

File: notebooks/assistants/function_calling/assistants_function_calling_with_bing_search.ipynb [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+      "Requirement already satisfied: annotated-types>=0.4.0 in c:\\users\\hakesava\\appdata\\local\\anaconda3\\envs\\trial\\lib\\site-packages (from pydantic<3,>=1.9.0->openai) (0.6.0)\n",
+      "Requirement already satisfied: pydantic-core==2.16.1 in c:\\users\\hakesava\\appdata\\local\\anaconda3\\envs\\trial\\lib\\site-packages (from pydantic<3,>=1.9.0->openai) (2.16.1)\n",
+      "Requirement already satisfied: colorama in c:\\users\\hakesava\\appdata\\local\\anaconda3\\envs\\trial\\lib\\site-packages (from tqdm>4->openai) (0.4.6)\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "#Install the packages\\\n",
+    "%pip install requests openai"
```
Comment (kdestin) [inline notebooks/assistants/function_calling/assistants_function_calling_with_bing_search.ipynb:38]: Add a loose version pin.

```suggestion
    "%pip install requests openai~=1.0"
```

**Review by kdestin — CHANGES_REQUESTED** [review]

File: notebooks/assistants/function_calling/assistants_function_calling_with_bing_search.ipynb (hunk 2 above)
Comment (hkesavam) [reply notebooks/assistants/function_calling/assistants_function_calling_with_bing_search.ipynb]: > Add a loose version pin.

added a pin

File: notebooks/assistants/function_calling/assistants_function_calling_with_bing_search.ipynb (hunk 1 above)
Comment (hkesavam) [reply notebooks/assistants/function_calling/assistants_function_calling_with_bing_search.ipynb]: I've tried doing this, the pre-commit checks erase it/fail.

**Review by kdestin — APPROVED** [review]

**Pull request merged**